flipperdevices/flipperzero-firmware
# [FL-3226] Deep Sleep Idle

# What's new

- Greatly improve battery life by going into STOP2 mode on idle.
- Add a setting to disable deep sleep (enabled by default).

# Verification 

1. Go to `Settings -> Power -> Battery Info`.
2. Wait until the backlight automatically turns off and observe the lower current consumption figures.
3. Go to `Settings -> System` and set `Sleep Method` to "Legacy". The current consumption should increase to its prior figures.
4. Set `Sleep Method` back to "Default".
5. Use Flipper normally. No regressions should be present.
6. (Optional) Use a meter to measure the current consumption directly.
7. (Optional) Set Flipper aside with no foreground apps running and note the time it takes to completely drain the battery. It must be several times greater than before.

# Checklist (For Reviewer)
